colossalai/checkpoint_io/hybrid_parallel_checkpoint_io.py

```diff
@@ -555,6 +575,25 @@ def save_lr_scheduler(self, lr_scheduler: LRScheduler, checkpoint: str):
         if self.coordinator.is_master():
             super().save_lr_scheduler(lr_scheduler, checkpoint)
 
+    def create_working_to_master_map(self, working_to_master_map: Mapping[Union[int, torch.Tensor], torch.Tensor]):
+        """
+        Create current checkpoint IO's working_to_master_map with passed in mapping.
+        This mapping can only be created when mixied precision is used.
+        The created mapping should be a mapping from address of working parameters to master parameter objects.
+
+        Args:
+            working_to_master_map (Mapping[Union[int, torch.Tensor], torch.Tensor]): A mapping from working parameters objects/addresses to master parameter objects.
+        """
+        self.working_to_master_map = dict()
+        for k, v in working_to_master_map.items():
+            if isinstance(k, torch.Tensor):
+                self.working_to_master_map[id(k)] = v
+            elif isinstance(k, int):
```

Comment:
When is it an integer?

Reply:
The mapping kept by LowLevelZeroOptimizer have working_param's addresses (id(working_param)) as keys, which are of integer type.  It can be checked in line50 of `colossalai/zero/low_level/bookkeeping/parameter_store.py`